designdocs/extended_token_lock.md

```diff
@@ -0,0 +1,162 @@
+# 📄 Boosting Rewards Extension
+
+## 📚 Context and Scope
+
+This design introduces a new boosting option in the Frequency runtime. It defines a mechanism for locking tokens that is both time-sensitive and event-driven, with immutable thawing behavior.
+
+While the core `Provider Boosting` system remains unchanged, this proposal builds upon it to incentivize long-term token commitments. Users who participate in extended locks will receive differentiated rewards via minimal extensions to the existing `capacity` pallet.
+
+## ❗ Problem Statement
+
+Business requirements call for a mechanism that allows users to:
+
+- Lock tokens before a **Precipitating Tokenomic Event (PTE)**.
+- Unlock tokens over time based on rules triggered by the PTE.
+- Maintain **immutability** of the unlock schedule—even governance cannot change it.
+- Reclaim tokens safely if the PTE never occurs (via a failsafe).
+- Earn **additional rewards** as compensation for committing to extended lock periods.
+
+### Additional Constraints
+
+- The PTE is a **Governance action**.
```

Comment:
what is the governance action over?

Reply:
Governance has to set the block number at which the event occurs, which starts the initial freeze period before the phased release. The PTE is a real-world event, and it requires a Governance action on-chain to record the block number at which it happened.